### ISSUE TITLE
Learning spatial relation wrt. area near the bounding box instead of the bounding box itself (yolo26n).

### ISSUE THREAD

**Issue by Shravasti221** [ultralytics/ultralytics#24981]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi all,

I had a question about the architecture of YOLO26n.

Suppose I'm training a detector with two classes: Human head, Human body
During training, does YOLO26n learn any explicit relationship between these classes (for example, that a head is usually spatially associated with a body), or are the predictions for each class learned independently, with only the shared backbone features providing contextual information?

I ask because I have tiny heads of 5x5 pixels which may not independently be sufficient to detect a head but the body is still identifiable. I do understand based on whatever I have read, the yolo architecture doesn't support class relations, but since the head bounding box is always close to a human body-like structure I was wondering about the possibility of the model being able to learn this.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Shravasti221, thank you for your interest in Ultralytics 🚀! This is an automated response to help get you moving quickly, and an Ultralytics engineer will also assist here soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
Since this is a custom training / model behavior ❓ question, please provide as much information as possible, including dataset image examples, label examples, training logs, your training command, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
If you’re able, please also share a small reproducible setup or validation example, as that can help us better understand the behavior you’re observing 🧩
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Shravasti221** [comment]
As a follow-up I would like to ask, that in case such a relation with a neighboring region is not possible, what is the possibility that the model will be able to learn these 5x5-10x10 pixel images in a landscape of size 1280x540? Is the ratio of the image to the bounding box size more important in determining if an object can be detected well enough or is it the absolute pixel size that the object occupies in the image a more important factor? I am seeing very poor map50 of 0.5 for the humanoid head so i am trying to pinpoint the root cause if it is an architectural limitation or there is some error in my training methodology

**Comment by Y-T-G** [comment]
It can learn relationships to other visual features. It can learn that a head is located near a feature that looks like a body.

The poor mAP is probably due to the small object size. The size is too small and it gets even smaller when the input is resized to fit imgsz. What matters is the object's absolute size when resized to imgsz.

5px would just be a single cell in the P3 feature map (80x80). It's so small that during down sampling, most of the details are lost.

**Comment by Shravasti221** [comment]
Would it be possible to guesstimate the range of improvement could I expect to see on using yolo26n-p2 model? I loaded the pre-trained weights from yolo26n.pt and then finetuned the yolo26n-p2 model on my dataset but my recall and map scores slipped. So I was hoping for some inputs. Since my model is for detecting body and head separately, I have been freezing the backbone (since I am detecting humanoid game characters) and training with the val results plateauing within 20/120 epochs (around 1% improvement over the next 50 epochs). I would like some help in understanding if this is expected or what could be the reason my training is plateauing this quickly.

I performed albumentation augmentations offline (dataset inflated with augmentations) and trained a yolo8n-p2. But when I brought in the same augmentations online I am not able to replicate or improve the results. Is there any reason that offline augmentations are more stable than on-the-fly ones?

Scanning /workspace/dataset/valid/labels.cache... 815 images, 5 backgrounds, 0 corrupt: 100% 
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 4.6it/s 11.4s
                   all        820       3186      0.887      0.799      0.852      0.619
                  self        148        148      0.906      0.973      0.967      0.878
               head        784       1465      0.837      0.624      0.715      0.375
               body        814       1573      0.918        0.8      0.873      0.604
Speed: 3.8ms preprocess, 7.3ms inference, 0.0ms loss, 0.1ms postprocess per image

Test set results:
               Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 788/788 34.8it/s 22.6s
                   all        788       3418      0.879      0.767      0.817      0.559
                  self        362        363      0.964       0.97      0.983      0.847
   enemy_head        773       1463      0.793      0.594      0.675      0.332
  enemy_body        774       1592      0.881      0.737      0.793      0.499

Below is the bounding box size distribution of my test set.

<img width="4234" height="3634" alt="Image" src="https://github.com/user-attachments/assets/1cba1008-32c4-423e-aff4-994e272991bb" />

<img width="4234" height="3634" alt="Image" src="https://github.com/user-attachments/assets/580e2d78-50bd-4cf5-a3fc-f2791b3b47af" />

In the train set I have 11,994 bounding boxes across classes and this is the bounding box distribution.

<img width="4234" height="3634" alt="Image" src="https://github.com/user-attachments/assets/1b069fc1-2444-4c37-a099-8fcef8998be3" />

<img width="4234" height="3634" alt="Image" src="https://github.com/user-attachments/assets/82c1da08-f72b-45f9-ade5-05704224f2a6" />

Could you possibly direct me to some resources that will help me understand what kind of expectations should I set for my metrics? I do see yolo models often performing at map50-90 = 90%+ very easily so I am trying to understand whether my expectations are not aligned due to dataset distribution, i.e. is this an acceptable number for the above test and train bounding box distribution or does the bounding box sizes point to lower map50 scores in the range of 70s for the head and similar scores for body? I think the larger question I am asking is what are the limitations to the nano model in terms of what it can learn. I do understand it is a very broad question but I was hoping to get reccomendations or resources that can help me come to a conclusion regarding whether there is some issue in my training methodology, dataset or the metrics are as expected.

**Comment by Y-T-G** [comment]
You can try YOLO11n-p2. You can use the following to transfer weights:

https://github.com/Y-T-G/community/discussions/60#discussioncomment-13425142

It's not possible to tell how much improvements you would see.

You can try disabling `mosaic` augmentations since your objects are small.

A better approach is probably to train a whole body pose model which detects body keypoints instead of a box.
